### PR TITLE
Closes #813: Create a tool that outputs the version info of a .dcf file

### LINKED ISSUE #813: Create a tool that outputs the version info of a .dcf file

### PR DESCRIPTION
**Pull request by rylin8**
Output the file name, ID, version, and timestamp. Also do this in the fetch tool.

### CODE REVIEW

File: crates/figma_import/src/bin/dcf_info.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+impl From<std::io::Error> for ConvertError {
+    fn from(e: std::io::Error) -> Self {
+        eprintln!("Error opening file: {:?}", e);
+        ConvertError(format!("Error opening file: {:?}", e))
+    }
+}
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[arg(short, long)]
```
Comment (timothyfroehlich) [inline crates/figma_import/src/bin/dcf_info.rs:37]: Could you add some basic usage statement here?

File: crates/figma_import/src/bin/dcf_info.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    dcf_file: std::path::PathBuf,
```
Comment (timothyfroehlich) [inline crates/figma_import/src/bin/dcf_info.rs:38]: Can you set the argument to be positional so we don't need to specify --dcf-file? Just remove the short and long traits. It's the only argument so I don't think it makes sense to require the flag

File: crates/figma_import/src/bin/dcf_info.rs
Comment (timothyfroehlich) [inline crates/figma_import/src/bin/dcf_info.rs:1]: Can you add a short bit about it in a readme? Either in figma_import's readme or in the main one. `cargo run --bin dcf_info --features="dcf_info -- --dcf-file <file>" is a mouthful.

**Review by timothyfroehlich — CHANGES_REQUESTED** [review]
I gave it a shot with the Tutorial file and it didn't work for some reason:

```
cargo run --bin dcf_info --features="dcf_info" -- --dcf-file reference-apps/tutorial/app/src/main/assets/figma/TutorialDoc_3z4xExq0INrL9vxPhj9tl7.dcf 
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/dcf_info --dcf-file reference-apps/tutorial/app/src/main/assets/figma/TutorialDoc_3z4xExq0INrL9vxPhj9tl7.dcf`
Error during deserialization: SizeLimit
dcf_info failed: ConvertError("Error during deserialization: SizeLimit")
```

**Comment by rylin8** [comment]
> I gave it a shot with the Tutorial file and it didn't work for some reason:
> 
> ```
> cargo run --bin dcf_info --features="dcf_info" -- --dcf-file reference-apps/tutorial/app/src/main/assets/figma/TutorialDoc_3z4xExq0INrL9vxPhj9tl7.dcf 
>     Finished dev [unoptimized + debuginfo] target(s) in 0.06s
>      Running `target/debug/dcf_info --dcf-file reference-apps/tutorial/app/src/main/assets/figma/TutorialDoc_3z4xExq0INrL9vxPhj9tl7.dcf`
> Error during deserialization: SizeLimit
> dcf_info failed: ConvertError("Error during deserialization: SizeLimit")
> ```

Oops, good catch. I was testing with a file retrieved from our fetch tool. I forgot this, but our fetch tool serializes and saves a file without the 4 byte version header, but files retrieved through live update do. So all of the dcf files saved for Validation and other apps have the header that I was not parsing.

I've updated this to parse the header and compare it with the current version. If they don't match, I print an error and abort since deserialization will fail. If they match I continue with deserialization of the rest of the file and print out the info.

File: crates/figma_import/src/bin/dcf_info.rs
Comment (rylin8) [reply crates/figma_import/src/bin/dcf_info.rs:1]: Done

File: crates/figma_import/src/bin/dcf_info.rs (hunk 2 above)
Comment (rylin8) [reply crates/figma_import/src/bin/dcf_info.rs:39]: Done

**Comment by iamralpht** [comment]
> Oops, good catch. I was testing with a file retrieved from our fetch tool. I forgot this, but our fetch tool serializes and saves a file without the 4 byte version header, but files retrieved through live update do. So all of the dcf files saved for Validation and other apps have the header that I was not parsing.
> 
> I've updated this to parse the header and compare it with the current version. If they don't match, I print an error and abort since deserialization will fail. If they match I continue with deserialization of the rest of the file and print out the info.

Ooh, it sounds like we generate different files from fetch versus live update. Is that still the case? (And if so, can we fix it in another PR?)

**Comment by rylin8** [comment]
> > Oops, good catch. I was testing with a file retrieved from our fetch tool. I forgot this, but our fetch tool serializes and saves a file without the 4 byte version header, but files retrieved through live update do. So all of the dcf files saved for Validation and other apps have the header that I was not parsing.
> > I've updated this to parse the header and compare it with the current version. If they don't match, I print an error and abort since deserialization will fail. If they match I continue with deserialization of the rest of the file and print out the info.
> 
> Ooh, it sounds like we generate different files from fetch versus live update. Is that still the case? (And if so, can we fix it in another PR?)

Yep https://github.com/google/automotive-design-compose/issues/816

**Comment by timothyfroehlich** [comment]
Oh and don't forget to try to avoid pushing amended commits. It makes it harder for me to pull down and test new changes.

**Review by iamralpht — APPROVED** [review]
Looks good, especially with the header fix landing, too.

**Comment by iamralpht** [comment]
> I gave it a shot with the Tutorial file and it didn't work for some reason:

@timothyfroehlich I think you can resolve your change request now, since we found/fixed the cause of the issue.

**Review by timothyfroehlich — APPROVED** [review]

**Pull request merged**